modules/broker-core/src/main/java/org/wso2/broker/core/Broker.java

```diff
@@ -64,7 +65,16 @@
     private BrokerHelper brokerHelper;
 
     public Broker(StartupContext startupContext) throws Exception {
-        this.messagingEngine = new MessagingEngine(startupContext.getService(DataSource.class));
+        MetricService metrics = startupContext.getService(MetricService.class);
+        if (Objects.nonNull(metrics)) {
```

Comment:
What does this check do?

Reply:
We thought it is self-explanatory. :open_mouth: What do you think it does?